### PR TITLE
Shape cli command

### PR DESCRIPTION
**Pull request by vladichhh**

### CODE REVIEW

File: cli-commands/shape/definition.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const nameOption = require('./options/name-option');
+
+module.exports = {
+    "template": "shape [name]",
```
Comment (bakasura980) [inline cli-commands/shape/definition.js]: shape [name] => shape [framework]

File: cli-commands/shape/definition.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const nameOption = require('./options/name-option');
```
Comment (bakasura980) [inline cli-commands/shape/definition.js]: name-option => framework-option

File: cli-commands/shape/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+const AsyncSoftExec = require('./../helpers/async-soft-exec');
+
+const Command = require('./../command');
+
+const commandMessages = require('./messages');
+const shapeCommandDefinition = require('./definition');
+
+const git = require('simple-git/promise')()
```
Comment (bakasura980) [inline cli-commands/shape/index.js]: const git = require('simple-git/promise')();

const Command = require('./../command');

const commandMessages = require('./messages');
const shapeCommandDefinition = require('./definition');

File: cli-commands/shape/index.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+const AsyncSoftExec = require('./../helpers/async-soft-exec');
```
Comment (bakasura980) [inline cli-commands/shape/index.js]: Don't need this

File: cli-commands/shape/index.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        super(shapeCommandDefinition);
+    }
+
+    async execute(args) {
+        try {
+            commandMessages.StartShaping();
+            const optionsResults = await super.processOptions(args);
+            const shape = optionsResults.name;
+
+            if (!shape.repository) {
```
Comment (bakasura980) [inline cli-commands/shape/index.js]: Don't need this check as if there is not such a shape => get the default one

File: cli-commands/shape/index.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+class ShapeCommand extends Command {
+    constructor() {
+        super(shapeCommandDefinition);
+    }
+
+    async execute(args) {
+        try {
+            commandMessages.StartShaping();
+            const optionsResults = await super.processOptions(args);
+            const shape = optionsResults.name;
```
Comment (bakasura980) [inline cli-commands/shape/index.js]: optionsResults.name; =>  optionsResults.framework;

File: cli-commands/shape/index.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+            if (!shape.repository) {
+                commandMessages.InvalidShapeName(shape.name);
+            }
+
+            await git.init();
```
Comment (bakasura980) [inline cli-commands/shape/index.js]: use `git.clone(optionsResults.framework)` instead of 3 lines pull

File: cli-commands/shape/index.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+            await git.addRemote('origin', shape.repository);
+            await git.pull('origin', 'master');
+
+            const asyncSoftExec = new AsyncSoftExec('npm install');
```
Comment (bakasura980) [inline cli-commands/shape/index.js]: Don't need to install dependencies for the project -> just download it and the user will go through the documentation to read how to do it himself

File: cli-commands/shape/messages.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+const chalk = require('chalk');
+
+module.exports = {
+    'StartShaping': () => { console.log(chalk.magentaBright('===== Shaping of DApp has started... =====')); },
+    'SuccessfulShaping': () => { console.log(chalk.greenBright(`===== Successful shaping of DApp =====`)); },
+    'UnsuccessfulShaping': (error) => {
+        console.log(chalk.redBright(`===== Unsuccessful shaping =====`));
+        console.log(error);
+    },
+    'InvalidShapeName': (name) => { console.log(chalk.redBright(`===== Invalid shape name ${name} =====`)); },
```
Comment (bakasura980) [inline cli-commands/shape/messages.js:10]: No needed

File: cli-commands/shape/messages.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+    'SuccessfulInstallation': () => { console.log(chalk.greenBright('===== Successfully installed project modules =====')); },
```
Comment (bakasura980) [inline cli-commands/shape/messages.js]: No needed

File: cli-commands/shape/messages.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    'UnsuccessfulInstallation': (error) => {
```
Comment (bakasura980) [inline cli-commands/shape/messages.js]: No needed

File: cli-commands/shape/options/name-option.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
+const Option = require('./../../option');
+const repositories = require('./../repositories.json');
+
+class NameOption extends Option {
+    constructor() {
+        super(
+            'name',
```
Comment (bakasura980) [inline cli-commands/shape/options/name-option.js]: name => framework

File: cli-commands/shape/options/name-option.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
+const Option = require('./../../option');
+const repositories = require('./../repositories.json');
+
+class NameOption extends Option {
```
Comment (bakasura980) [inline cli-commands/shape/options/name-option.js]: NameOption => FrameworkOption

File: cli-commands/shape/options/name-option.js [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+            {
+                "describe": "The name of the framework that the project will be build up.",
+                "type": "string",
+                "default": "react"
+            }
+        );
+    }
+
+    process(optionValue) {
+        return {
```
Comment (bakasura980) [inline cli-commands/shape/options/name-option.js]: `return repositories[optionValue] || repositories['react'];`

File: cli-commands/shape/repositories.json [hunk 15]
```diff
@@ -0,0 +1,3 @@
+{
+    "REACT": "https://github.com/LimeChain/eoslime-shape-react.git"
```
Comment (bakasura980) [inline cli-commands/shape/repositories.json]: REACT => react

File: cli.js [hunk 16]
```diff
@@ -16,11 +17,13 @@ const MochaFramework = require('./cli-commands/test/test-frameworks/mocha');
     const testCommand = new TestCommand();
     const deployCommand = new DeployCommand();
     const compileCommand = new CompileCommand();
+    const shapeCommand = new ShapeCommand();
```
Comment (bakasura980) [inline cli.js]: move it between other rows above so the final view will be ordered by lenght
...
.....
.......
..........

File: cli.js [hunk 17]
```diff
@@ -5,6 +5,7 @@ const InitCommand = require('./cli-commands/init');
 const TestCommand = require('./cli-commands/test');
 const DeployCommand = require('./cli-commands/deploy');
 const CompileCommand = require('./cli-commands/compile');
+const ShapeCommand = require('./cli-commands/shape');
```
Comment (bakasura980) [inline cli.js]: move it between other rows above so the final view will be ordered by lenght
...
.....
.......
..........

File: package.json [hunk 18]
```diff
@@ -17,6 +17,7 @@
         "is-invalid-path": "^1.0.2",
         "mocha": "5.2.0",
         "prompts": "^2.2.1",
+        "simple-git": "^1.132.0",
```
Comment (bakasura980) [inline package.json]: Please remove all `^` so we will use the version without an automatic upgrade. I have missed removing other `^`

**Review by bakasura980 — CHANGES_REQUESTED** [review]

File: cli-commands/shape/index.js (hunk 8 above)
Comment (vladichhh) [reply cli-commands/shape/index.js]: removed installation of project dependencies

File: cli-commands/shape/messages.js (hunk 10 above)
Comment (vladichhh) [reply cli-commands/shape/messages.js]: removed

File: cli-commands/shape/messages.js (hunk 11 above)
Comment (vladichhh) [reply cli-commands/shape/messages.js]: removed

File: cli-commands/shape/options/name-option.js (hunk 12 above)
Comment (vladichhh) [reply cli-commands/shape/options/name-option.js]: updated property name

File: cli-commands/shape/repositories.json (hunk 15 above)
Comment (vladichhh) [reply cli-commands/shape/repositories.json]: updated

File: cli-commands/shape/options/name-option.js (hunk 13 above)
Comment (vladichhh) [reply cli-commands/shape/options/name-option.js]: updated

File: cli-commands/shape/definition.js (hunk 2 above)
Comment (vladichhh) [reply cli-commands/shape/definition.js]: updated

File: cli-commands/shape/definition.js (hunk 1 above)
Comment (vladichhh) [reply cli-commands/shape/definition.js]: updated

File: cli-commands/shape/index.js (hunk 4 above)
Comment (vladichhh) [reply cli-commands/shape/index.js]: removed

File: cli-commands/shape/index.js (hunk 3 above)
Comment (vladichhh) [reply cli-commands/shape/index.js]: order updated

File: cli-commands/shape/index.js (hunk 6 above)
Comment (vladichhh) [reply cli-commands/shape/index.js]: updated

File: cli.js (hunk 17 above)
Comment (vladichhh) [reply cli.js]: updated

File: cli.js (hunk 16 above)
Comment (vladichhh) [reply cli.js]: updated

File: package.json (hunk 18 above)
Comment (vladichhh) [reply package.json]: ^ removed

File: cli-commands/shape/definition.js [hunk 19]
```diff
@@ -0,0 +1,7 @@
+const frameworkOption = require('./options/framework-option');
+
+module.exports = {
+    "template": "shape [framework]",
+    "description": "Shapes a ready to use DApp, based on specified framework",
+    options: [frameworkOption]
```
Comment (bakasura980) [inline cli-commands/shape/definition.js]: options => "options"

File: cli-commands/shape/messages.js [hunk 20]
```diff
@@ -0,0 +1,11 @@
+const chalk = require('chalk');
+
+module.exports = {
+    'StartShaping': () => { console.log(chalk.magentaBright('===== Shaping of DApp has started... =====')); },
+    'SuccessfulShaping': () => { console.log(chalk.greenBright(`===== Successful shaping of DApp =====`)); },
```
Comment (bakasura980) [inline cli-commands/shape/messages.js]: "Successful shaping" only

File: cli-commands/shape/options/framework-option.js [hunk 21]
```diff
@@ -0,0 +1,21 @@
+const Option = require('../../option');
+const repositories = require('./../repositories.json');
+
+class FrameworkOption extends Option {
+    constructor() {
+        super(
+            'framework',
+            {
+                "describe": "The name of the framework that the project will be build up.",
```
Comment (bakasura980) [inline cli-commands/shape/options/framework-option.js]: "The name of the framework that the project will be build up." => "A front-end framework a project will be prepared for you on"

**Review by bakasura980 — CHANGES_REQUESTED** [review]

File: cli-commands/shape/definition.js (hunk 19 above)
Comment (vladichhh) [reply cli-commands/shape/definition.js]: fixed

File: cli-commands/shape/messages.js (hunk 20 above)
Comment (vladichhh) [reply cli-commands/shape/messages.js]: updated

**Review by bakasura980 — APPROVED** [review]

**Pull request merged**